op-node/rollup/derive/frame_test.go

```diff
@@ -23,3 +31,122 @@ func FuzzParseFrames(f *testing.F) {
 		}
 	})
 }
+
+func TestFrameMarshaling(t *testing.T) {
```

Comment:
I think this would be more suitable to fuzz testing. I had a problem in the past where the rng was deterministic when running `go test`

Reply:
I also wanted to go with fuzzing first, but Go offers very limited control over how fuzzy data is generated. If you look at `randomFrame`, it covers all possible max values for e.g. the frame number and data length. To the best of my knowledge, I couldn't easily achieve this with go's fuzzing capabilities, so I decided to write the custom random frame generator. Also note that there are already two general fuzz tests at the top of the `frame_test.go` file.

I followed the example in other tests in the monorepo and created a random frame generator `randomFrame` that takes a `rng`, which is seeded with the unix nano time (`time.Now().UnixNano()`) at the start of each test. This should provide sufficiently random seeds and thus sufficiently random generated frames. I'm aware of instances in the past (e.g. [here](https://github.com/golang/go/issues/22037) or [here](https://groups.google.com/g/golang-dev/c/DboHw1eCiSc)) where, on MacOS, repeated calls returned the same value. But if test runs are even milliseconds apart, this isn't a problem. And the latest the CI would run with a new seed every time.